subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -112,10 +124,20 @@ void bt_ots_dir_list_obj_remove(struct bt_ots_dir_list *dir_list, void *obj_mana
 	}
 
 	while (offset < dir_list->net_buf.len) {
-		uint16_t len = sys_get_le16(dir_list->net_buf.data + offset);
-		uint64_t id = sys_get_le64(dir_list->net_buf.data + offset + sizeof(len));
+		uint16_t len;
+		uint64_t id;
+
+		__ASSERT((DIR_LIST_OBJ_RECORD_MIN_SIZE + offset) <= dir_list->net_buf.len,
+			 "Invalid dir_list buf length %u, expected at least %u",
+			 dir_list->net_buf.len, DIR_LIST_OBJ_RECORD_MIN_SIZE + offset);
+
+		len = sys_get_le16(dir_list->net_buf.data + offset);
+		id = sys_get_le64(dir_list->net_buf.data + offset + sizeof(len));
 
 		__ASSERT(len, "Invalid object length");
+		__ASSERT((len + offset) <= dir_list->net_buf.len,
+			 "Invalid dir_list buf length %u, expected at least %u",
+			 dir_list->net_buf.len, len + offset);
```

Comment:
Does `dir_list.net_buf` originate from some remote peer device? If so, then assert is the wrong choice here. That should only be used to catch local programming bugs, whereas invalid data from a peer should be handled gracefully.

Reply:
The dir_list.net_buf (in fact everything related to the dir_list) is created by the OTS service locally. It is an internal read-only object. 

The dir_list object is, however, created based on metadata from other, potentially remotely created objects, but my opinion is that it is the responsibility of the OTS implementation (and not the OTS dir list implementation) to validate remotely created objects. That being said, the current version of our OTS implementation does not allow a client to create objects, but only the local application.